_posts/2024-11-08-navigating-the-ecosystem.markdown

```diff
@@ -0,0 +1,72 @@
+---
+layout: post
+title:  "Finding your way around the CHERIoT ecosystem"
+date:   2024-11-07
+categories: rtos sonata git
+author: Phil Day
+---
+With a new set of folks in the DSbD cohort 6 just getting started with their Sonata boards I thought it might be useful to capture some of the things I've learn in the last few months about the ecosystem.
+
+### Setting up a dev environment
+The easiest and quickest way to get up and running is to use the dev container.
+I have a fairly strong aversion to installing new software on my base system, and this just requires that you have docker installed.
+Opening a repo with dev container support in VS code and it will ask you if you want to reopen in a container.
+Do that and in a few seconds you're in a pre-build dev environment.
+Simples, to paraphrase a meerkat.
+
+One word of warning though; cheriot and Sonata both uses the ":latest" tag as the version of the docker image, so you won't automatically get updates when they are releases.
+If you've pulled an update to a repo and builds starts failing with no apparent reason it's worth pulling the latest dev container image and relaunching the dev container.
+You can find the image name to pull in the .devcontainer/devcontainer.json file.
+
+### Which Repo do I use ?
+
+The CHERIoT ecosystem is split across a number of git repositories, and makes extensive used of [submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) to link them and other external code from other places such as FreeRTOS.
+There is no version or release tagging in the repos, so you can only tell how current a submodule is by comparing the commit hashes.
+On the plus side the CI system and maintainers do a good job of keeping it all compatible, and any breaking changes are discussed in the Signal channel.
+
+#### [cheriot-rtos](https://github.com/CHERIoT-Platform/cheriot-rtos.git)
+Cheriot-rtos is the core of the system, an includes its own set of examples that provide a great introduction to the main CHERI features.
+You don't even need any hardware to run them, as by default then build for the ibex-safe-simulator that's included in dev container.
```

Comment:
Most of them default to building for the Sail model, the Ibex model has working input on the UART so a few use that.  You mostly don't need to notice this because `xmake run` does the right thing.